kork/kork-artifacts/src/test/java/com/netflix/spinnaker/kork/artifacts/artifactstore/entities/DeserializerHookRegistryTest.java

```diff
@@ -0,0 +1,186 @@
+                        "float": 3.14,
+                        "bool": true,
+                        "object": {
+                            "string": "child",
+                            "int": 2,
+                            "float": 2.58,
+                            "bool": false
+                        },
+                        "array_simple": [0, 1, 2],
+                        "array_complex": [
+                            {
+                                "elem": 0,
+                                "name": "array_0"
+                            },
+                            {
+                                "elem": 1,
+                                "name": "array_1"
+                            },
+                            {
+                                "elem": 2,
+                                "name": "array_2"
+                            }
+                        ]
+                    }
+                    """,
+            Map.of(
+                "string",
+                "parent",
+                "int",
+                1,
+                "float",
+                3.14,
+                "bool",
+                true,
+                "object",
+                Map.of("string", "child", "int", 2, "float", 2.58, "bool", false),
+                "array_simple",
+                List.of(0, 1, 2),
+                "array_complex",
+                List.of(
+                    Map.of("elem", 0, "name", "array_0"),
+                    Map.of("elem", 1, "name", "array_1"),
+                    Map.of("elem", 2, "name", "array_2")))),
+        Arguments.of(
+            """
+                   {
+                    "stringValue": "foo",
+                    "intValue": 5,
+                    "floatValue": 3.14,
+                    "boolValue": true,
+                    "listValue": ["v1","v2","v3"],
+                    "intArrayValue": [3,2,1],
+                    "child": {
+                        "childValue": "c1"
+                    }
+                   }
+                   """,
+            new MockClass(
+                "foo",
+                5,
+                3.14f,
+                true,
+                List.of("v1", "v2", "v3"),
+                new Integer[] {3, 2, 1},
+                new NestedMockClass("c1"))),
+        Arguments.of(
+            """
+                 {
+                     "name": "foo"
+                 }
+                """,
+            new JsonCreatorAnnotationCheck("foo")),
+        Arguments.of(
+            """
+                     {
+                         "manifests": [
+                            {
+                              "type": "remote/map/base64",
+                              "reference": "ref://application/hash"
+                            }
+                         ]
+                     }
+                    """,
+            Map.of("manifests", List.of(Map.of("hello", "world!")))));
+  }
+
+  @SneakyThrows
```

Comment:
Is this better than adding `throws Exception` to the methd signature.  Seems like that's pretty typical for test methods.